src/treetensornetworks/solvers/tree_sweeping.jl

```diff
@@ -63,3 +63,117 @@ function half_sweep(dir::Base.ReverseOrdering, args...; kwargs...)
     reverse(half_sweep(Base.Forward, args...; kwargs...)),
   )
 end
+
+function default_region_args_func(internal_kwargs::NamedTuple, pre_region_args::NamedTuple)
+  #wrap args in another NamedTuple
+  internal_kwargs = merge(get(pre_region_args, :internal_kwargs, (;)), internal_kwargs)
+  return merge(pre_region_args, (; internal_kwargs))
+end
+
+function default_sweep_plan(
+  nsites,
+  graph::AbstractGraph;
+  region_args_func=default_region_args_func,
+  reverse_args_func=default_region_args_func,
+  pre_region_args=(;),
+  kwargs...,
+)  ###move this to a different file, algorithmic level idea
+  return vcat(
+    [
+      half_sweep(
```

Comment:
Rename `forward_sweep(...)` or something similar.

Remove the direction, and define a reverse sweep as `reverse(forward_sweep(...))`.

Reply:
This will also require fixing places in the code which rely on the direction of edges (currently the `inserter` function does for example, but it should be rewritten using `region` and `next_region` logic), but that should of course be done anyway.